lib/vSphere/action/get_ssh_info.rb

```diff
@@ -13,21 +13,27 @@ def initialize(app, _env)
 
         def call(env)
           env[:machine_ssh_info] = get_ssh_info(env[:vSphere_connection], env[:machine])
-
           @app.call env
         end
 
         private
 
+        def filter_guest_nic(vm, machine)
+          return vm.guest.ipAddress unless machine.provider_config.real_nic_ip
+          adapters = vm.guest.net.select { |g| g.deviceConfigId > 0 }.map { |g| g.ipAddress[0] }
```

Comment:
It looks like the `map` call here will end up giving us an array of IP addresses, not adapter objects. If I'm reading this correctly, `adapters` should be renamed to `ips` or something.